src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt

```diff
@@ -406,11 +369,30 @@ class CodyAutocompleteManager {
     val instance: CodyAutocompleteManager
       get() = service()
 
-    @JvmStatic
-    fun diff(a: String, b: String): Patch<String> =
-        DiffUtils.diff(characterList(a), characterList(b))
+    private val lineBreaks = listOf("\r\n", "\n", "\r")
+
+    @VisibleForTesting
+    fun trimCommonPrefixAndSuffix(completion: String, original: String): Pair<Int, String> {
+      var startIndex = 0
+      var endIndex = completion.length
+
+      // Trim common prefix
+      while (startIndex < completion.length &&
+          startIndex < original.length &&
+          completion[startIndex] == original[startIndex]) {
+        startIndex++
+      }
+
+      // Trim common suffix
+      while (endIndex > 0 &&
+          endIndex > startIndex &&
+          original.length - (completion.length - endIndex) > 0 &&
```

Comment:
I think it would be simpler this way:

```
var endIndex = completion.length - 1
while (endIndex > 0 &&
          endIndex > startIndex &&
          completion[endIndex] ==
              original[original.length - (completion.length - endIndex)]) {
        endIndex--
}
```

Moving `-1` out will make sure `endIndex > 0` condition is correct.
`original.length - (completion.length - endIndex) > 0` condition should be redundant if ` endIndex > startIndex`, no?